llvm/include/llvm/SandboxIR/Context.h

```diff
@@ -48,6 +61,21 @@ class Context {
   /// Type objects.
   DenseMap<llvm::Type *, std::unique_ptr<Type, TypeDeleter>> LLVMTypeToTypeMap;
 
+  /// Callbacks called when an IR instruction is about to get removed. Keys are
+  /// used as IDs for deregistration.
+  DenseMap<int, RemoveInstrCallback> RemoveInstrCallbacks;
```

Comment:
Do you want to keep them as ints or:
```
using ID = int;
```
?

Reply:
Good point, thanks! Switched it to `CallbackID`.